platform/android/src/snapshotter/map_snapshotter.cpp

```diff
@@ -79,16 +83,24 @@ void MapSnapshotter::start(JNIEnv&) {
             static auto onSnapshotReady = javaClass.GetMethod<void (jni::Object<MapSnapshot>)>(*_env, "onSnapshotReady");
             javaPeer->Call(*_env, onSnapshotReady, mapSnapshot);
         }
+        activationCounter--;
+        if(activationCounter==0){
+            jFileSource->pause(*_env);
+        }
     });
 
     snapshotter->snapshot(snapshotCallback->self());
 }
 
-void MapSnapshotter::cancel(JNIEnv&) {
+void MapSnapshotter::cancel(JNIEnv& env) {
     MBGL_VERIFY_THREAD(tid);
     snapshotCallback.reset();
-}
 
+    if (activationCounter > 0) {
```

Comment:
Can we simplify this to a boolean in the MapSnapshotter? Don't think we need an actual counter in this case right?

Reply:
This is needed because the MapSnapshotter can be reused for generating mulitple maps. When start is called we increase by one, when we receive a callback we decrease by one [here](https://github.com/mapbox/mapbox-gl-native/pull/10556/files#diff-71ad6c747ca48b85bf4f30842aa0a2a7R86). If the end developer calls into cancel (eg. as part of destroying the activity) we are resetting the counter. A boolean doesn't give us an indication if we have ongoing requests, eg. generating multiple maps in one activity would stop working after generating one bitmap.